engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -149,17 +153,18 @@ public final HSQuery targetedTypes(List<CustomTypeMetadata> types) {
 			return targetedEntities( null );
 		}
 
-		List<Class<?>> entityTypes = types.stream()
+		List<IndexedTypeIdentifier> entityTypes = types.stream()
 				.map( CustomTypeMetadata::getEntityType ).collect( toList() );
```

Comment:
I agree but let's do after the next phase: I'm pretty confident that with another general pass we wouldn't even have a `List<CustomTypeMetadata>` as method input ;)